hikari/internal/override.py

```diff
@@ -0,0 +1,42 @@
+# cython: language_level=3
```

Comment:
Could we rename this file to `typing_shim`, so that we can include more stuff, like deprecations, if we need to add runtime compatibility for stuff from `typing_extensions`?